network/wsNetwork.go

```diff
@@ -1975,8 +1972,18 @@ func (wn *WebsocketNetwork) numOutgoingPending() int {
 // GetHTTPClient returns a http.Client with a suitable for the network Transport
 // that would also limit the number of outgoing connections.
 func (wn *WebsocketNetwork) GetHTTPClient(address string) (*http.Client, error) {
+	url, err := addr.ParseHostOrURL(address)
+	if err != nil {
+		return nil, err
+	}
+
+	maxIdleConnsPerHost := int(wn.config.ConnectionsRateLimitingCount)
```

Comment:
Why was this moved out of setup?

Reply:
because it is now "bound" transport to a specific target (one extra required argument) so it is created per peer, and not reusing a shared generic instance